src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java

```diff
@@ -291,10 +303,47 @@ public void onBlockDropItems(BlockDropItemEvent e) {
     @EventHandler
     public void onBlockBreak(BlockBreakEvent e) {
         if (SlimefunTag.CAVEMAN_TALISMAN_TRIGGERS.isTagged(e.getBlock().getType())) {
-            Talisman.checkFor(e, SlimefunItems.TALISMAN_CAVEMAN);
+            Talisman.tryActivate(e, SlimefunItems.TALISMAN_CAVEMAN);
+        }
+    }
+
+    @EventHandler
+    public void onExperienceReceive(PlayerExpChangeEvent e) {
+        if (e.getAmount() > 0 && Talisman.tryActivate(e, SlimefunItems.TALISMAN_WISE)) {
+            e.setAmount(e.getAmount() * 2);
+        }
+    }
+
+    @EventHandler
+    public void onPlayerDeath(PlayerDeathEvent e) {
+        Player player = e.getEntity();
+        DamageCause dmgCause = player.getLastDamageCause().getCause();
+        ItemStack talisman = Talisman.tryActivateAndGet(e, SlimefunItems.TALISMAN_RESURRECTED);
+
+        if (dmgCause == DamageCause.VOID && talisman != null) {
+            // This event is cancelled by the Talisman, so they don't actually die, we just teleport them after 2 ticks.
+            SlimefunPlugin.runSync(() -> PaperLib.teleportAsync(player, getSafeRespawnLocation(talisman, player)), 2);
```

Comment:
Any reason you delay this 2 ticks? This event will keep firing as the player is on half a heart and keeps getting damage from the void.